packages/oslo-generator-shacl-template/lib/handlers/UniqueLanguageConstraintHandler.ts

```diff
@@ -0,0 +1,93 @@
+import { getApplicationProfileLabel, Logger, ns, type QuadStore } from "@oslo-flanders/core";
+import type * as RDF from '@rdfjs/types';
+import { GenerationMode } from "@oslo-generator-shacl-template/enums/GenerationMode";
+import { TranslationKey } from "@oslo-generator-shacl-template/enums/TranslationKey";
+import type { NamedOrBlankNode } from "@oslo-generator-shacl-template/types/IHandler";
+import { ShaclHandler } from "@oslo-generator-shacl-template/types/IHandler";
+import { TranslationService } from "@oslo-generator-shacl-template/TranslationService";
+import { ShaclTemplateGenerationServiceConfiguration } from "@oslo-generator-shacl-template/config/ShaclTemplateGenerationServiceConfiguration";
+import { ShaclTemplateGenerationServiceIdentifier } from "@oslo-generator-shacl-template/config/ShaclTemplateGenerationServiceIdentifier";
+import { inject } from "inversify";
+
+export class UniqueLanguageConstraintHandler extends ShaclHandler {
+  public constructor(
+    @inject(ShaclTemplateGenerationServiceIdentifier.Configuration) config: ShaclTemplateGenerationServiceConfiguration,
+    @inject(ShaclTemplateGenerationServiceIdentifier.Logger) logger: Logger,
+    @inject(ShaclTemplateGenerationServiceIdentifier.TranslationService) translationService: TranslationService,
+  ) {
+    super(config, logger, translationService);
+  }
+
+  public handle(
+    subject: NamedOrBlankNode,
+    store: QuadStore,
+    shaclStore: QuadStore,
+  ): void {
+    const range: RDF.NamedNode | undefined = <RDF.NamedNode | undefined>store.findObject(subject, ns.rdfs('range'));
+
+    if (!range) {
+      throw new Error(`Unable to find the range for subject "${subject.value}".`);
+    }
+
+    const rangeAssignedURI = store.getAssignedUri(range);
+
+    if (!rangeAssignedURI) {
+      throw new Error(`Unable to find the assigned URI for range "${range.value}".`);
+    }
+
+    if (rangeAssignedURI.equals(ns.rdf('langString'))) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
```js
// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
const shapeId: NamedOrBlankNode = this.propertyIdToShapeIdMap.get(subject.value)!;
```
This is something I see returning a couple of times. Maybe there is a better way to handle this?